### PR TITLE
Feature: Automate release flow

### LINKED ISSUE #30: Automate release flow
Depends on:
* https://github.com/getsentry/sentry-godot/issues/43

In Sentry we use [craft](https://github.com/getsentry/craft) for release.
And the [publish repo](https://github.com/getsentry/publish) for approval workflow.

Details on how to set up `publish` and `craft` are on the readme of those repos.

CI should create a single artifact that will be used by craft. So if we're building the package as a result of Mac+Linux+Windows we might need to set up CI in a way that a final job grabs all the OS-specific artifacts and creates a single zip at the end.

This can create the GitHub Release, create the release registry entry, and run any other target. We need to create a new craft target for AssetLib. We can start with GH release only, and AssetLib becomes a goal closer to a more stable/complete version of the SDK.

The trigger workflow is like this: https://github.com/getsentry/sentry-unity/blob/main/.github/workflows/release.yml

The version is bumped is done via a script. Craft calls a `.sh` but we pass that to the `ps1`:
https://github.com/search?q=repo%3Agetsentry%2Fsentry-dotnet%20bump-version&type=code

### PR DESCRIPTION
**Pull request by limbonaut**
Included:
- Bump version script
- Craft configuration
- Package workflow
- Release workflow

Closes #30

### CODE REVIEW

**Comment by limbonaut** [comment]
I think this is as close as I could get without actually testing with real tokens and creating a release. `craft` seems to fail with no tags existing in the repo (`git describe` fails), so I've been adding a simple tag on the first commit in my tests. This looks like an oversight in craft, but the workaround is trivial.

I can also add registry to the configuration:
```yaml
  - name: registry
    sdks:
      github:getsentry/sentry-godot:
```
I'm not sure what else is needed for the registry to work.

File: .github/workflows/package.yml
Comment (bruno-garcia) [inline .github/workflows/package.yml:22]: Ideally this goes into a script so it's easier to test locally and debug stuff

File: .github/workflows/release.yml
Comment (bruno-garcia) [inline .github/workflows/release.yml:22]: I verified these are set in the repo already

File: CHANGELOG.md
Comment (bruno-garcia) [inline CHANGELOG.md:1]: Add:

### Unreleased

This will get replaced with the version

File: scripts/bump-version.ps1
Comment (bruno-garcia) [inline scripts/bump-version.ps1:1]: I believe craft looks for `.sh` only, that's why we had a .sh that calls into powershell.
Unless that changed.

Oh but since we explicitly use the command on the craft.yml that might work. Lets try

File: .github/workflows/package.yml
Comment (limbonaut) [reply .github/workflows/package.yml:22]: It relies on actually downloading an artifact - something that is a merged product of a number of builds. Testing it locally would be cumbersome, since it's not enough to produce a single build for this.

File: CHANGELOG.md
Comment (limbonaut) [reply CHANGELOG.md:1]: Fixed

File: scripts/bump-version.ps1
Comment (limbonaut) [reply scripts/bump-version.ps1:1]: I adopted this approach from `sentry-unreal`.

**Comment by limbonaut** [comment]
This is a test build run in my fork: <https://github.com/limbonaut/sentry-godot/actions/runs/12354096820>

File: .github/workflows/package.yml
Comment (bruno-garcia) [reply .github/workflows/package.yml:22]: I meant the steps. There are a few commands here that could go in a `.sh` file and we invoke it from here.

File: CHANGELOG.md
Comment (bruno-garcia) [reply CHANGELOG.md:1]: Not showing on the diff yet, did you push it?

File: .craft.yml
Comment (bruno-garcia) [inline .craft.yml:4]: Since we're not using release registry here, this PR is unblocked.

On a follow up PR, we can add a new target, like this:
https://github.com/getsentry/sentry-unreal/blob/4027d198c6ab87016d95665fcf9af1df3cac4a31/.craft.yml#L12-L13

But basically as:

```
  - name: registry
    sdks:
      github:getsentry/sentry-godot:
```


That is, once this is merged: https://github.com/getsentry/sentry-release-registry/pull/174

**Review by bruno-garcia — APPROVED** [review]

**Comment by bruno-garcia** [comment]
I merged to test it out since it requires approval anyway once a release is triggered.

Triggering [here](https://github.com/getsentry/sentry-godot/actions/workflows/release.yml) as 0.0.1 (since we're not pushing to the release registry yet, since I already created 0.0.1 there).

Run: https://github.com/getsentry/sentry-godot/actions/runs/12404236789/job/34629077725

But failed with:

```
[info] Releasing version 0.0.1 from main
[info] Preparing to release the version: 0.0.1
[info] Created a new release branch: "release/0.0.1"
[info] Switched to branch "release/0.0.1"
Error:  fatal: No names found, cannot describe anything.

  
  at Object.action (/usr/local/bin/craft:357:10622)
  at fPe.exec (/usr/local/bin/craft:357:10934)
  at /usr/local/bin/craft:356:6689
  at new Promise (<anonymous>)
  at bS.handleTaskData (/usr/local/bin/craft:356:6578)
  at bS.<anonymous> (/usr/local/bin/craft:356:6204)
  at Generator.next (<anonymous>)
  at a (/usr/local/bin/craft:353:1502)
  at process.processTicksAndRejections (node:internal/process/task_queues:[95](https://github.com/getsentry/sentry-godot/actions/runs/12404236789/job/34629077725#step:4:101):5)
Error: Process completed with exit code 1.
```

Seems to be described on this issue:
* https://github.com/getsentry/craft/issues/342

> this is not so simple -- the prepare scripts pass the previous version string to the script hooks. so there must be something for those. the [quick start guide](https://develop.sentry.dev/sdk/craft-quick-start/) currently recommends tagging the first commit as 0.0.0

I'll create a dummy tag: https://github.com/getsentry/sentry-godot/releases/tag/0.0.0 


Rerunning the workflow: https://github.com/getsentry/sentry-godot/actions/runs/12404236789

Approval issue created: https://github.com/getsentry/publish/issues/4791
Approved

version going from 0.1.0-dev to 0.0.1 which is fine for this test: https://github.com/getsentry/sentry-godot/compare/2ea6be34ab5c4ae6e8ba4c23c63f821ca23bdff3...refs/heads/release/0.0.1


Release logs: https://github.com/getsentry/publish/actions/runs/12404320280/job/34629296353

Failed: https://github.com/getsentry/publish/issues/4791#issuecomment-2552588334
Need to debug this

**Comment by limbonaut** [comment]
> Failed: [getsentry/publish#4791 (comment)](https://github.com/getsentry/publish/issues/4791#issuecomment-2552588334) Need to debug this

See comment https://github.com/getsentry/publish/issues/4791#issuecomment-2554895265

**Pull request merged**
